rewriter/flatten.cc

```diff
@@ -161,6 +161,10 @@ class FlattenWalk {
             }
             moveQueue[idx] = {move(expr), staticLevel};
         }
+
+        bool atTopLevel() const {
+            return stack.empty();
+        }
```

Comment:
We  use this to determine whether to replace a moved `MethodDef` with a symbol or an `EmptyTree`. (This isn't strictly necessary, but it makes the exp files a lot cleaner.)

Reply:
Let's make a comment that this is _syntactical_ top level rather than semantical and add a tests for 
```
puts(def foo; end;)
```
on top level

Reply:
(also, ~every comment left in PRs to explain something is worth leaving in source)